---
 usb.ids |  177 ++++++++++++++++++++++++++++++++++++++++++++++++++++------------
 1 file changed, 144 insertions(+), 33 deletions(-)

--- a/usb.ids
+++ b/usb.ids
@@ -9,8 +9,8 @@
 #	The latest version can be obtained from
 #		http://www.linux-usb.org/usb.ids
 #
-# Version: 2011.09.28
-# Date:    2011-09-28 20:34:04
+# Version: 2011.11.13
+# Date:    2011-11-13 20:34:03
 #
 
 # Vendors, devices and interfaces. Please keep sorted.
@@ -150,7 +150,7 @@
 	0111  G55xi Printer/Scanner/Copier
 	0117  LaserJet 3200
 	011c  hn210w 802.11b Adapter
-	011d  Integrated Bluetooth Module
+	011d  Bluetooth 1.2 Interface [Broadcom BCM2035]
 	0121  HP49g+ Calculator
 	0122  HID Internet Keyboard
 	0201  ScanJet 6200c
@@ -177,7 +177,7 @@
 	0411  OfficeJet G95
 	0412  Printing Support
 	0417  LaserJet 1200 series
-	0423  HS-COMBO
+	0423  HS-COMBO Cardreader
 	0504  DeskJet 885c
 	0505  ScanJet 2100c
 	0507  DVD+RW
@@ -185,7 +185,7 @@
 	0511  OfficeJet K60
 	0512  DeckJet 450
 	0517  LaserJet 1000
-	051d  Wireless (Bluetooth + WLAN) Interface [Integrated Module]
+	051d  Bluetooth Interface
 	0601  ScanJet 6300c
 	0604  DeskJet 840c
 	0605  ScanJet 2200c
@@ -271,7 +271,7 @@
 	1711  psc 780xi
 	1712  Printing Support
 	1717  LaserJet 3020
-	171d  Wireless (Bluetooth + WLAN) Interface [Integrated Module]
+	171d  Bluetooth 2.0 Interface [Broadcom BCM2045]
 	1801  Inkjet P-2000U
 	1802  PhotoSmart 470 series
 	1804  DeskJet 916C
@@ -279,7 +279,7 @@
 	1811  PSC 720
 	1812  OfficeJet Pro K550
 	1817  LaserJet 3030
-	181d  integrated module with Bluetooth 2.0 wireless technology.
+	181d  Bluetooth 2.0 Interface
 	1902  PhotoSmart A430 series
 	1904  DeskJet 3820
 	1911  OfficeJet V45
@@ -447,6 +447,7 @@
 	4d17  EWS UPD
 	4e11  PhotoSmart 2570 series
 	4f11  OfficeJet 5600 (USBHUB)
+	4f17  Color LaserJet CM1312 MFP
 	5004  DeskJet 995c
 	5011  PhotoSmart 3100 series
 	5017  EWS UPD
@@ -562,6 +563,7 @@
 	9b02  PhotoSmart M540 series
 	9c02  PhotoSmart M440 series
 	a004  DeskJet 5850c
+	a011  Deskjet 3050A
 	b002  PhotoSmart 7200 series
 	b102  PhotoSmart 7200 series
 	b116  Webcam
@@ -996,6 +998,7 @@
 	019e  WLI-UC-GNP Wireless LAN Adapter
 	01a1  MiniStation Metro
 	01a2  WLI-UC-GNM Wireless LAN Adapter
+	01ee  WLI-UC-GNM2 Wireless LAN Adapter [Ralink RT3070]
 0412  Award Software International
 0413  Leadtek Research, Inc.
 	1310  WinFast TV - NTSC + FM
@@ -1158,6 +1161,7 @@
 	4068  Live! Cam Notebook [VF0470]
 	406c  Live! Cam Sync [VF0520]
 	4083  Live! Cam Socialize [VF0640]
+	4087  Live! Cam Socialize HD 1080 [VF0680]
 	4100  Nomad Jukebox 2
 	4101  Nomad Jukebox 3
 	4102  NOMAD MuVo^2
@@ -1680,6 +1684,7 @@
 	005c  Enhanced Laser Gaming Device
 	005d  Enhanced Device
 	0061  Bluetooth Dongle
+	0066  Genius Traveler 1000 Wireless Mouse
 	0072  Navigator 335
 	0083  Bluetooth Dongle
 	0087  Ergo 525V Laser Mouse
@@ -2135,6 +2140,7 @@
 	0810  QuickCam Pro
 	0819  Webcam C210
 	081b  Webcam C310
+	081d  HD Webcam C510
 	0820  QuickCam VC
 	0821  HD Webcam C910
 	0825  Webcam C270
@@ -2296,12 +2302,14 @@
 	c053  Laser Mouse
 	c058  M115 Mouse
 	c05a  Optical Mouse M90
+	c05b  M-U0004 810-001317 [B110 Optical USB Mouse]
 	c05d  Optical Mouse
 	c061  RX1500 Laser Mouse
 	c062  LS1 Laser Mouse, corded
 	c063  DELL Laser Mouse
 	c068  G500 Laser Mouse
 	c06b  G700 Wireless Gaming Mouse
+	c06c  Optical Mouse
 	c101  UltraX Media Remote
 	c110  Harmony 885 Remote
 	c111  Harmony 525 Remote
@@ -2328,6 +2336,7 @@
 	c21a  Precision Gamepad
 	c21c  G13 Advanced Gameboard
 	c21d  F310 Gamepad [XInput Mode]
+	c21e  F510 Gamepad [XInput Mode]
 	c21f  F710 Wireless Gamepad [XInput Mode]
 	c221  G11/G15 Keyboard / Keyboard
 	c222  G15 Keyboard / LCD
@@ -2710,6 +2719,7 @@
 	481d  BT Digital Access adapter
 	5000  ST Micro/Ergenic ERG BT-002 Bluetooth Adapter
 	5001  ST Micro Bluetooth Device
+	5710  Joystick in FS Mode
 	5721  Hantek DDS-3X25 Arbitrary Waveform Generator
 	7270  ST Micro Serial Bridge
 	7554  56k SoftModem
@@ -3339,6 +3349,7 @@
 	312d  Elura 100
 	3138  PowerShot A710 IS
 	3147  EOS 1Ds Mark III
+	314f  Powershot SD1000
 	3155  PowerShot A450
 	315a  PowerShot G9
 	315d  PowerShot A720
@@ -3347,6 +3358,7 @@
 	3176  PowerShot A590
 	317a  Powershot A470
 	3184  Digital IXUS 80 IS (PTP mode)
+	3192  PowerShot SX110 IS
 	319a  EOS 7D
 	31bc  PowerShot D10
 	31bf  PowerShot A480
@@ -3356,6 +3368,8 @@
 	31ef  PowerShot A495
 	31f3  PowerShot Digital ELPH SD1400 IS
 	31ff  Digital IXUS 55
+	3210  Powershot SX30 IS
+	3214  SELPHY CP800
 	3229  IXUS 220 HS
 04aa  DaeWoo Telecom, Ltd
 04ab  Chromatic Research
@@ -3835,6 +3849,7 @@
 	01f7  FinePix J250 (PTP)
 	01fd  A160
 	023e  FinePix AX300
+	0240  FinePix S2950 Digital Camera
 04cc  ST-Ericsson
 	1122  Hub
 	1520  USB 2.0 Hub (Avocent KVM)
@@ -3851,6 +3866,7 @@
 	04ce  SL11DEMO, VID: 0x4ce, PID: 0x4ce
 	07d1  SL11R, VID: 0x4ce, PID: 0x07D1
 04cf  Myson Century, Inc.
+	0022  OCZ Alchemy Series Elixir II Keyboard
 	0800  MTP800 Mass Storage Device
 	8810  CS8810 Mass Storage Device
 	8811  CS8811 Mass Storage Device
@@ -3901,6 +3917,7 @@
 	1400  PS/2 keyboard + mouse controller
 	1503  Shortboard Lefty
 	1603  Keyboard
+	2013  Keyboard [Das Keyboard]
 	2221  Keyboard
 04da  Panasonic (Matsushita)
 	0901  LS-120 Camera
@@ -4324,6 +4341,7 @@
 	b036  Asus Integrated 0.3M UVC Webcam
 	b044  Acer CrystalEye Webcam
 	b057  integrated USB webcam
+	b059  CKF7037 HP webcam
 	b071  2.0M UVC Webcam / CNF7129
 	b091  Webcam
 	b104  CNF7069 Webcam
@@ -4335,6 +4353,7 @@
 	b1cf  Lenovo Integrated Camera
 	b221  integrated camera
 	b230  Integrated HP HD Webcam
+	b272  Lenovo EasyCamera
 04f3  Elan Microelectronics Corp.
 	01a4  Wireless Keyboard
 	0210  AM-400 Hama Optical Mouse
@@ -4385,6 +4404,7 @@
 	002b  HL-5250DN Printer
 	002c  Printer
 	002d  Printer
+	0039  HL-5340 series
 	0100  MFC8600/9650 series
 	0101  MFC9600/9870 series
 	0102  MFC9750/1200 series
@@ -4542,6 +4562,7 @@
 	2015  QL-500 P-touch label printer
 	2016  QL-550 P-touch label printer
 	201a  PT-18R P-touch label printer
+	201b  QL-650TD P-Touch Label Printer
 	2027  QL-560 P-Touch Label Printer
 	2100  Card Reader Writer
 04fa  Dallas Semiconductor
@@ -4659,6 +4680,7 @@
 	0249  USB 2 Flash Media Device
 	0257  F5U257 Serial
 	0304  FSU304 USB 2.0 - 4 Ports Hub
+	0307  USB 2.0 - 7 ports Hub [FSU307]
 	0409  F5U409 Serial
 	0551  F6C550-AVR UPS
 	0706  2-N-1 USB 2.0 7-Port Hub (Lower half)
@@ -4873,6 +4895,7 @@
 0547  Anchor Chips, Inc.
 	0001  ICSI Bluetooth Device
 	1002  Python2 WDM Encoder
+	1006  Hantek DSO-2100 UF
 	2131  AN2131 EZUSB Microcontroller
 	2235  AN2235 EZUSB-FX Microcontroller
 	2710  EZ-Link Loader (EZLNKLDR.SYS)
@@ -5422,8 +5445,9 @@
 	5601  AVM Fritz!WLAN [Texas Instruments TNETW1450]
 	6201  AVM Fritz!WLAN v1.1 [Texas Instruments TNETW1450]
 	62ff  AVM Fritz!WLAN USB (in CD-ROM-mode)
-	8401  AVM Fritz!WLAN N [Atheros AR9001U]
-	8402  AVM Fritz!WLAN N 2.4 [Atheros AR9001U]
+	8401  Fritz!WLAN N [Atheros AR9001U]
+	8402  Fritz!WLAN N 2.4 [Atheros AR9001U]
+	8403  Fritz!WLAN N v2 [Atheros AR7010]
 	84ff  AVM Fritz!WLAN USB N (in CD-ROM-mode)
 057d  Shark Multimedia, Inc.
 057e  Nintendo Co., Ltd
@@ -5745,6 +5769,7 @@
 	0061  Varo PocketZip 40 MP3 Player
 	006d  HipZip MP3 Player
 	007c  Ultra Max USB/1394
+	007d  HTC42606 0G9AT00 [Iomega HDD]
 	007e  Mini 256MB/512MB Flash Drive [IOM2D5]
 	00db  FotoShow Zip 250 Driver
 	0150  Mass Storage Device
@@ -5880,6 +5905,18 @@
 	0236  Internal Keyboard/Trackpad (ANSI)
 	0237  Internal Keyboard/Trackpad (ISO)
 	0238  Internal Keyboard/Trackpad (JIS)
+	023f  Internal Keyboard/Trackpad (ANSI)
+	0240  Internal Keyboard/Trackpad (ISO)
+	0241  Internal Keyboard/Trackpad (JIS)
+	0242  Internal Keyboard/Trackpad (ANSI)
+	0243  Internal Keyboard/Trackpad (ISO)
+	0244  Internal Keyboard/Trackpad (JIS)
+	0245  Internal Keyboard/Trackpad (ANSI)
+	0246  Internal Keyboard/Trackpad (ISO)
+	0247  Internal Keyboard/Trackpad (JIS)
+	0252  Internal Keyboard/Trackpad (ANSI)
+	0253  Internal Keyboard/Trackpad (ISO)
+	0254  Internal Keyboard/Trackpad (JIS)
 	0301  USB Mouse [Mitsumi, M4848]
 	0302  Optical Mouse [Fujitsu]
 	0304  Optical USB Mouse [Mitsumi]
@@ -5947,6 +5984,7 @@
 	8216  Bluetooth USB Host Controller
 	8217  Bluetooth USB Host Controller
 	8218  Bluetooth Host Controller
+	821a  Bluetooth Host Controller
 	8240  IR Receiver [built-in]
 	8241  IR Receiver [built-in]
 	8242  IR Receiver [built-in]
@@ -5956,6 +5994,7 @@
 	8502  Built-in iSight
 	8505  Built-in iSight
 	8507  Built-in iSight
+	8509  FaceTime HD Camera
 	911c  Hub in A1082 [Cinema HD Display 23"]
 	912f  Hub in 30" Cinema Display
 	921c  A1082 [Cinema HD Display 23"]
@@ -6317,7 +6356,7 @@
 05e0  Symbol Technologies
 	0700  Bar Code Scanner (CS1504)
 	0800  Spectrum24 Wireless LAN Adapter
-	1200  DS6608 Bar Code Scanner
+	1200  Bar Code Scanner
 	1900  SNAPI Imaging Device
 	2000  MC3090 Rugged Mobile Computer
 	200d  MC70 Rugged Mobile Computer
@@ -6540,6 +6579,7 @@
 	0102  Wireless Keyboard/Mouse Combo [MK1152WC]
 	0201  Defender Office Keyboard (K7310) S Zodiak KM-9010
 	0252  Emerge Uni-retractable Laser Mouse
+	3286  Nano Receiver [Sandstrom Laser Mouse SMWLL11]
 	9003  VoIP Conference Hub (A16GH)
 	9004  USR9602 USB Internet Mini Phone
 062b  Greatlink Electronics Taiwan, Ltd
@@ -6556,14 +6596,16 @@
 	0003  Vivicam 35Xx
 0638  Avision, Inc.
 	0268  iVina 1200U Scanner
-	026a  Minolta Dimage Scan Dual II
+	026a  Minolta Dimage Scan Dual II AF-2820U (2886)
 	0a10  iVina FB1600/UMAX Astra 4500
 	0a13  AV600U
+	0a15  Konica Minolta SC-110
 	0a16  SC-215
 	0a30  UMAX Astra 6700 Scanner
 	0a41  Avision AM3000/MF3000 Series
 	0f01  fi-4010CU
-	4004  Minolta Dimage Scan Elite II
+# typo?
+	4004  Minolta Dimage Scan Elite II AF-2920 (2888)
 0639  Chrontel, Inc.
 063a  Techwin Corp.
 063b  Taugagreining HF
@@ -6891,7 +6933,7 @@
 	3034  Develop D 2050iD PCL
 	4001  Dimage 2300
 	4003  Dimage 2330 Zoom Camera
-	4004  Scan Elite II
+	4004  Dimage Scan Elite II AF-2920 (2888)
 	4005  Minolta DiMAGE E201 Mass Storage Device
 	4006  Dimage 7 Camera
 	4007  Dimage S304 Camera
@@ -6900,8 +6942,8 @@
 	400a  Dimage S404 Camera
 	400b  Dimage 7i Camera
 	400c  Dimage F100 Camera
-	400d  Scan Dual III
-	400e  Dimage 5400
+	400d  Dimage Scan Dual III AF-2840 (2889)
+	400e  Dimage Scan Elite 5400 (2890)
 	400f  Dimage 7Hi Camera
 	4010  Dimage Xi Camera
 	4011  Dimage F300 Camera
@@ -7507,6 +7549,7 @@
 	4588  XBox Device
 	8818  Street Fighter IV Arcade FightStick (PS3)
 073a  Chaplet Systems, Inc.
+	2230  infrared dongle for remote
 073b  Suncom Technologies
 073c  Industrial Electronic Engineers, Inc.
 	0305  Pole Display (PC305-3415  2 x 20 Line Display)
@@ -7596,6 +7639,7 @@
 	201a  M-Audio Micro
 	201b  M-Audio RunTime DFU
 	201d  M-Audio Producer
+	2024  M-Audio Fast Track MKII
 	2080  M-Audio RunTime DFU
 	2081  M-Audio RunTime DFU / Fast Track Ultra 8R
 	2803  M-Audio Audiophile DFU
@@ -8187,28 +8231,28 @@
 	13ed  VvBus for Helium 2xx
 	13f1  DSL-302G Modem
 	13f2  DSL-502G Router
-	3300  DWA-130 802.11n Wireless N Adapter(rev.E) [Realtek RTL8192SU]
-	3302  DWA-130 802.11n Wireless N Adapter(rev.C2) [Realtek RTL8192SU]
+	3300  DWA-130 802.11n Wireless N Adapter(rev.E) [Realtek RTL8191SU]
+	3302  DWA-130 802.11n Wireless N Adapter(rev.C2) [Realtek RTL8191SU]
 	3303  DWA-131 802.11n Wireless N Nano Adapter(rev.A1) [Realtek RTL8192SU]
 	3a07  WUA-2340 RangeBooster G Adapter(rev.A) [Atheros AR5523]
 	3a08  WUA-2340 RangeBooster G Adapter(rev.A) (no firmware) [Atheros AR5523]
 	3a09  DWA-160 802.11abgn Xtreme N Dual Band Adapter(rev.A2) [Atheros AR9170+AR9104]
 	3a0d  DWA-120 802.11g Wireless 108G Adapter [Atheros AR5523]
-	3a0f  DWA-130 802.11n Wireless N Adapter(rev.D) [Atheros AR9001U-(2)NG]
+	3a0f  DWA-130 802.11n Wireless N Adapter(rev.D) [Atheros AR9170+AR9102]
 	3a10  DWA-126 802.11n Wireless Adapter [Atheros AR9271]
 	3b01  AirPlus G DWL-G122 Wireless Adapter(rev.D) [Marvell 88W8338+88W8010]
 	3b10  RangeBooster N Adapter
-	3b11  DWA-130 802.11n Wireless N Adapter(rev.A1) [Marvell W8360USB]
-	3c03  AirPlus G DWL-G122 Wireless Adapter(rev.C1) [Ralink RT73]
+	3b11  DWA-130 802.11n Wireless N Adapter(rev.A1) [Marvell 88W8362+88W8060]
+	3c03  AirPlus G DWL-G122 Wireless Adapter(rev.C1) [Ralink RT2571W]
 	3c04  WUA-1340
 	3c05  EH103 Wireless G Adapter
 	3c07  Wireless G DWA-110 Adapter
 	3c09  DWA-140 RangeBooster N Adapter(rev.B1) [Ralink RT2870]
-	3c0a  DWA-140 RangeBooster N Adapter(rev.B2) [Ralink RT2870]
+	3c0a  DWA-140 RangeBooster N Adapter(rev.B2) [Ralink RT3072]
 	3c0b  DWA-110 Wireless G USB Adapter(rev.B)
 	3c0d  DWA-125 Wireless N 150 Adapter(rev.A1) [Ralink RT3070]
 	3c0e  WUA-2340 RangeBooster G Adapter(rev.B) [Ralink RT2070]
-	3c0f  AirPlus G DWL-G122 Wireless Adapter(rev.E) [Ralink RT2870]
+	3c0f  AirPlus G DWL-G122 Wireless Adapter(rev.E1) [Ralink RT2070]
 	3c10  DWA-160 802.11abgn Xtreme N Dual Band Adapter(rev.A1) [Atheros AR9170+AR9104]
 	3c11  DWA-160 Xtreme N Dual Band USB Adapter(rev.B) [Ralink RT2870]
 	3c13  DWA-130 802.11n Wireless N Adapter(rev.B) [Ralink RT2870]
@@ -8617,6 +8661,14 @@
 	2902  PCM2902 Audio Codec
 	2904  PCM2904 Audio Codec
 	2910  PCM2912 Audio Codec
+	29b0  PCM2900B Audio CODEC
+	29b2  PCM2902 Audio CODEC
+	29b3  PCM2903B Audio CODEC
+	29b6  PCM2906B Audio CODEC
+	29c0  PCM2900C Audio CODEC
+	29c2  PCM2902C Audio CODEC
+	29c3  PCM2903C Audio CODEC
+	29c6  PCM2906C Audio CODEC
 08bd  Citizen Watch Co., Ltd
 	0208  CLP-521 Label Printer
 	1100  X1-USB Floppy
@@ -9017,6 +9069,7 @@
 	130b  F3507g Mobile Broadband Module
 	130c  F3607gw Mobile Broadband Module
 	1311  F3607gw v2 Mobile Broadband Module
+	1400  Memory Stick 2GB
 	642f  TravelDrive
 	6506  TravelDrive 2C
 	6507  TravelDrive 2C
@@ -9073,6 +9126,7 @@
 	0030  Composite Device, Mass Storage Device (Flash Drive) amd HID
 	003c  Rhythmedics HID Bootloader
 0939  Lumberg, Inc.
+	0b15  Toshiba Stor.E Alu 2 1TB (PX1710E-1HJ0)
 093a  Pixart Imaging, Inc.
 	0007  CMOS 100K-R Rev. 1.90
 	010e  Digital camera, CD302N/Elta Medi@ digi-cam/HE-501A
@@ -9821,6 +9875,7 @@
 0b0d  ProjectLab
 	0000  CenturyCD
 0b0e  GN Netcom
+	1022  Jabra PRO 9450, Type 9400BS (DECT Headset)
 0b0f  AVID Technology
 0b10  Pcally
 0b11  I Tech Solutions Co., Ltd
@@ -10380,6 +10435,7 @@
 	0012  Bluetooth Device(BC04-External)
 	0018  Bluetooth Device(BC04-External)
 	0019  Bluetooth Device
+	0021  Bluetooth Device
 	0c24  Bluetooth Device(SAMPLE)
 	ffff  Bluetooth module with BlueCore in DFU mode
 0c25  Sampo Corp.
@@ -11357,6 +11413,7 @@
 0e8d  MediaTek Inc.
 	0003  MT6227 phone
 	0004  MT6227 phone
+	1836  Samsung SE-S084 Super WriteMaster Slim External DVD writer
 0e8f  GreenAsia Inc.
 	0003  MaxFire Blaze2
 	0012  USB Wireless 2.4GHz Gamepad
@@ -11589,7 +11646,8 @@
 0fb1  Inclose Design, Inc.
 0fb2  Juan-Chern Industrial Co., Ltd
 0fb6  Heber Ltd
-	3fc4  Firefly X10i USB I/O Board
+	3fc3  Firefly X10i I/O Board (with firmware)
+	3fc4  Firefly X10i I/O Board (without firmware)
 0fb8  Wistron Corp.
 	0002  eHome Infrared Receiver
 0fb9  AACom Corp.
@@ -11708,7 +11766,7 @@
 1001  Ritronics Components (S) Pte., Ltd
 1003  Sigma Corp.
 	0003  SD14
-	0100  Sigma SD10
+	0100  SD9/SD10
 1004  LG Electronics, Inc.
 	1fae  U8120 3G Cellphone
 	6000  KU330/KU990/VX4400/VX6000
@@ -11852,6 +11910,7 @@
 	0704  Passport External HDD
 	070a  My Passport Essential SE
 	071a  My Passport 1TB
+	0740  My Passport 1TB
 	0742  My Passport Essential SE
 	0900  MyBook Essential External HDD
 	0901  MyBook External HDD
@@ -12396,6 +12455,7 @@
 	1035  U8120
 	1037  Ideos
 	1038  Ideos (debug mode)
+	1039  Ideos (tethering mode)
 	1406  E1750
 	140b  EC1260 Wireless Data Modem HSD USB Card
 	1446  E1552/E1800 (HSPA modem)
@@ -12427,7 +12487,7 @@
 	0163  256MB/512MB/1GB Flash Drive
 	0165  2GB/4GB Flash Drive
 	0330  63-in-1 Multi-Card Reader/Writer
-	0361  CR-75: 51-in-1 Card Reader/Writer
+	0361  CR-75: 51-in-1 Card Reader/Writer [Sakar]
 	1169  TS2GJF210 JetFlash 210 2GB
 	1171  Fingerprint Reader
 1308  Shuttle, Inc.
@@ -12448,12 +12508,12 @@
 	0006  Dimage Z2 Camera
 	0007  Minolta DiMAGE Z2 (PictBridge mode)
 	0008  Dimage X21 Camera
-	000a  Dimage Scan Dual IV
+	000a  Dimage Scan Dual IV AF-3200 (2891)
 	000b  Dimage Z10 Camera
 	000d  Dimage X50 Camera [storage?]
 	000f  Dimage X50 Camera [p2p?]
 	0010  Dimage G600 Camera
-	0012  Dimage Scan Elite5400 2
+	0012  Dimage Scan Elite 5400 II (2892)
 	0013  Dimage X31 Camera
 	0015  Dimage G530 Camera
 	0017  Dimage Z3 Camera
@@ -12564,8 +12624,11 @@
 	0023  WUSB54GR
 	0024  WUSBF54G v1.1 802.11bg
 	0026  WUSB54GSC v1 802.11g Adapter [Broadcom 4320 USB]
+	0028  WUSB200 802.11g Adapter [Ralink RT2671]
 	002f  AE1000 v1 802.11n [Ralink RT3572]
 	0031  AM10 v1 802.11n [Ralink RT3072]
+	0039  AE1200 802.11bgn Wireless Adapter [Broadcom BCM43235]
+	003a  AE2500 802.11abgn Wireless Adapter [Broadcom BCM43236]
 	13b1  WUSB200: Wireless-G Business Network Adapter with Rangebooster
 13b2  Alesis
 	0030  Multimix 8
@@ -12635,6 +12698,7 @@
 13e1  Kaibo Wire & Cable (Shenzhen) Co., Ltd.
 13e5  Rane
 	0001  SL-1
+	0003  TTM 57SL
 13e6  TechnoScope Co., Ltd.
 13ea  Hengstler
 	0001  C-56 Thermal Printer
@@ -12656,6 +12720,7 @@
 	1f00  DataTraveler 2.0 4GB Flash Drive / Patriot Xporter 32GB (PEF32GUSB) Flash Drive
 	2240  microSD card reader
 	3100  2/4 GB stick
+	3800  Rage XT Flash Drive
 1400  Axxion Group Corp.
 1402  Bowe Bell & Howell
 1403  Sitronix
@@ -12724,6 +12789,7 @@
 	013f  Megapixel Auto Focus Webcam
 	0142  WB-6250X Webcam
 	015a  WB-8300X 2MP Webcam
+	0167  Widescreen 3MP Webcam
 1460  Tatung Co.
 	9150  eHome Infrared Transceiver
 1461  Staccato Communications
@@ -12788,6 +12854,7 @@
 14af  ATP Electronics Inc.
 14b0  StarTech.com Ltd.
 14b2  Ralink Technology, Corp.
+	3a93  Topcom 802.11bg Wireless Adapter [Atheros AR5523]
 	3a95  Toshiba WUS-G06G-JT 802.11bg Wireless Adapter
 	3a98  Airlink101 AWLL4130 802.11bg Wireless Adapter [Atheros AR5523]
 	3c02  rt2500 802.11g WLAN
@@ -12797,6 +12864,7 @@
 	3c09  802.11n adapter
 	3c22  Conceptronic C54RU 802.11bg Wireless Adapter [Ralink RT73]
 	3c23  Airlink101 AWLL6080 802.11bgn Wireless Adapter [Ralink RT2870]
+	3c24  NEC NP01LM 802.11abg Wireless Adapter [Ralink RT2571W]
 	3c25  DrayTek Vigor N61 802.11bgn Wireless Adapter [Ralink RT2870]
 	3c27  Airlink101 AWLL6070 802.11bgn Wireless Adapter [Ralink RT2770]
 	3c28  Conceptronic C300RU v2 802.11bgn Wireless Adapter [Ralink RT2770]
@@ -12898,7 +12966,7 @@
 1570  ALLTOP TECHNOLOGY CO., LTD.
 157b  Ketron SRL
 157e  TRENDnet
-	3006  TEW-444UB EU
+	3006  TEW-444UB EU [TRENDnet]
 	3007  TEW-444UB EU (no firmware)
 	300a  TEW-429UB 802.11bg
 	300b  TEW-429UB 802.11bg
@@ -13345,6 +13413,7 @@
 	9801  EUB9801 802.11abgn Wireless Adapter [Ralink RT3572]
 1743  General Atomics
 174c  ASMedia Technology Inc.
+	5106  Transcend StoreJet 25M3
 174f  Syntek
 	1105  SM-MS/Pro-MMC-XD Card Reader
 	1403  Integrated Webcam
@@ -13416,9 +13485,10 @@
 	0004  Linux-USB Midi Gadget
 17ba  SAURIS GmbH
 	0001  SAU510-USB [no firmware]
-	0510  SAU510-USB and SAU510-USP plus JTAG Emulators
+	0510  SAU510-USB and SAU510-USB plus JTAG Emulators
 	0511  SAU510-USB Iso Plus JTAG Emulator
 	0520  SAU510-USB Nano JTAG Emulator
+	1511  Onboard Emulator on SAUModule development kit
 17c3  Singim International Corp.
 17cc  Native Instruments
 	041c  Audio 2 DJ
@@ -13539,6 +13609,7 @@
 	4e21  Nexus S
 	4e22  Nexus S (debug)
 	4e24  Nexus S (tether)
+	7102  Toshiba Thrive tablet
 18d5  Starline International Group Limited
 18d9  Kaba
 	01a0  B-Net 91 07
@@ -13693,13 +13764,16 @@
 1a0a  USB-IF non-workshop
 	badd  USB OTG Compliance test device
 1a12  KES Co., Ltd.
+1a1d  Veho
+	0407  Mimi WiFi speakers
 1a25  Amphenol East Asia Ltd.
 1a2a  Seagate Branded Solutions
 1a32  Quanta Microsystems, Inc.
 	0304  802.11n Wireless LAN Card
 1a36  Biwin Technology Ltd.
-1a40  TERMINUS TECHNOLOGY INC.
-	0101  USB-2.0 4-Port HUB
+1a40  Terminus Technology Inc.
+	0101  4-Port HUB
+	0201  Hub
 1a41  Action Electronics Co., Ltd.
 1a44  VASCO Data Security International
 	0001  Digipass 905 SmartCard Reader
@@ -13716,13 +13790,16 @@
 	0068  VerticalMouse 3
 	0168  VerticalMouse 3 Wireless
 	0191  VerticalMouse 4
+1a81  Holtek Semiconductor, Inc.
+	2203  Laser Gaming mouse
+	2204  Optical Mouse
 1a86  QinHeng Electronics
 	5523  CH341 in serial mode, usb to serial port converter
 	5584  CH341 in parallel mode, usb to printer port converter
 	7523  HL-340 USB-Serial adapter
 	752d  CH345 MIDI adapter
 	7584  CH340S
-	e008  HID-based USB-serial converter, full-speed, similar to HE2325U
+	e008  HID-based serial adapater
 1a89  Dynalith Systems Co., Ltd.
 1a8b  SGS Taiwan Ltd.
 1a8d  BandRich, Inc.
@@ -13733,8 +13810,11 @@
 1aa4  Data Drive Thru, Inc.
 1aa5  UBeacon Technologies, Inc.
 1aa6  eFortune Technology Corp.
+1ab1  Rigol Technologies
+	0588  DS1000 SERIES
 1acb  Salcomp Plc
 1ad1  Desay Wire Co., Ltd.
+1adb  SEL C662 Serial Cable
 1ae4  ic-design Reinhard Gottinger GmbH
 1ae7  X-TENSIONS
 	0381  VS-DVB-T 380U (af9015 based)
@@ -13818,6 +13898,12 @@
 	810b  ME-8100B
 	820a  ME-8200A
 	820b  ME-8200B
+1b0e  BLUTRONICS S.r.l.
+	1078  BLUDRIVE II CCID
+1b1c  Corsair
+	0890  Flash Padlock
+	0a00  SP2500 Speakers
+	1a90  Flash Voyager GT
 1b20  MStar Semiconductor, Inc.
 1b22  WiLinx Corp.
 1b26  Cellex Power Products, Inc.
@@ -13863,6 +13949,9 @@
 	1000  xHC1 Controller
 1b75  Ovislink Corp.
 	3072  AirLive WN-360USB adapter
+	8187  AirLive WL-1600USB 802.11g Adapter [Realtek RTL8187L]
+	9170  Airlive X.USB [AR9170]
+	a200  AirLive WN-200USB wireless 11b/g/n dongle
 1b76  Legend Silicon Corp.
 1b80  Afatech
 	c810  MC810 [af9015]
@@ -13874,6 +13963,7 @@
 	e383  DVB-T UB383-T [af9015]
 	e385  DVB-T UB385-T [af9015]
 	e386  DVB-T UB385-T [af9015]
+	e399  DVB-T KWorld PlusTV 399U [af9015]
 	e39a  DVB-T395U [af9015]
 	e39b  DVB-T395U [af9015]
 	e409  IT9137FN Dual DVB-T [KWorld UB499-2T]
@@ -13904,9 +13994,12 @@
 1bce  Contac Cable Industrial Limited
 1bcf  Sunplus Innovation Technology Inc.
 	0007  Optical Mouse
+	053a  Targa Silvercrest OMC807-C optische Funkmaus
 	05c5  SPRF2413A [2.4GHz Wireless Keyboard/Mouse Receiver]
 	05cf  Micro keyboard & mouse receiver
+	0c31  Serial-ATA bridge
 1bd0  Hangzhou Riyue Electronic Co., Ltd.
+1bd5  BG Systems, Inc.
 1bde  P-TWO INDUSTRIES, INC.
 1bef  Shenzhen Tongyuan Network-Communication Cables Co., Ltd
 1bf0  RealVision Inc.
@@ -13940,11 +14033,19 @@
 1c37  Authorizer Technologies, Inc.
 1c3d  NONIN MEDICAL INC.
 1c3e  Wep Peripherals
+1c40  EZPrototypes
+	0533  TiltStick
+	0534  i2c-tiny-usb interface
+	0535  glcd2usb interface
+	0536  Swiss ColorPAL
 1c49  Cherng Weei Technology Corp.
 1c4f  SiGma Micro
+	0002  Keyboard TRACER Gamma Ivory
 	0003  HID controller
+	000e  Genius KB-120 Keyboard
 	3000  Micro USB Web Camera
 1c6b  Philips & Lite-ON Digital Solutions Corporation
+	a222  DVD Writer Slimtype eTAU108
 1c6c  Skydigital Inc.
 1c73  AMT
 	861f  Anysee E30 USB 2.0 DVB-T Receiver
@@ -13952,12 +14053,17 @@
 1c78  Datascope Corp.
 1c79  Unigen Corporation
 1c7a  LighTuning Technology Inc.
+	0801  Fingerprint Reader
 1c7b  LUXSHARE PRECISION INDUSTRY (SHENZHEN) CO., LTD.
 1c87  2N TELEKOMUNIKACE a.s.
 1c88  Somagic, Inc.
+	0007  SMI Grabber (EasyCAP DC60+ clone) (no firmware) [SMI-2021CBE]
+	003c  SMI Grabber (EasyCAP DC60+ clone) [SMI-2021CBE]
 1c89  HONGKONG WEIDIDA ELECTRON LIMITED
 1c8e  ASTRON INTERNATIONAL CORP.
 1c98  ALPINE ELECTRONICS, INC.
+1c9e  OMEGA TECHNOLOGY
+	6061  WL-72B 3.5G MODEM
 1ca0  ACCARIO Inc.
 1cac  Kinstone
 	a332  C8 Webcam
@@ -14012,11 +14118,12 @@
 	0103  NCM (Ethernet) Gadget
 	0104  Multifunction Composite Gadget
 	0105  FunctionFS Gadget
-	0106  Composite Gadget: ACM + Mass Storage
 	0200  Qemu Audio Device
 1e0e  Qualcomm / Option
 1e10  Point Grey Research, Inc.
 	2004  Sony 1.3MP 1/3" ICX445 IIDC video camera [Chameleon]
+1e17  Mirion Technologies Dosimetry Services Division
+	0001  instadose dosimeter
 1e1d  Lumension Security
 	0165  Secure Pen drive
 1e1f  INVIA
@@ -14035,6 +14142,8 @@
 	040a  LPC2378 [Robotino Bootloader]
 	0501  CP2102 [CMSP]
 	0601  CMMP-AS
+1e3d  Chipsbank Microelectronics Co., Ltd
+	4082  CBM4082 SD Card Reader
 1e41  Cleverscope
 	0001  CS328A PC Oscilloscope
 1e54  TypeMatrix
@@ -14054,6 +14163,8 @@
 1ebb  NuCORE Technology, Inc.
 1eda  AirTies Wireless Networks
 	2310  802.11n USB Wireless LAN Card
+1edb  Blackmagic design
+	bd3b  Intensity Shuttle
 1ee8  ONDA COMMUNICATION S.p.a.
 	0014  MT833UP
 1ef6  EADS Deutschland GmbH
